### PR TITLE
db/climate-hamburger

### PR DESCRIPTION
**Pull request by karolkolodziej**
- [x] Fahrenheit

### CODE REVIEW

File: samples/dashboard/demos/climate/demo.js [hunk 1]
```diff
@@ -537,7 +538,42 @@ async function setupDashboard() {
...
+                items: [
+                    'editMode',
+                    {
+                        id: 'dark-mode',
+                        type: 'toggle',
+                        text: 'Dark mode',
+                        events: {
+                            click: function () {
+                                const dashboard = this.menu.editMode.dashboard,
+                                    darModeClass = 'dark-mode';
```
[the remark this answers is not in the document]
Comment (pawelfus) [reply samples/dashboard/demos/climate/demo.js:560]: I agree with your suggestion above: prefix + className, as we do in Highcharts.

**Comment by pawelfus** [comment]
Note:
This PR causes loading sources in sync (instead of lazy-loading)

File: samples/dashboard/demos/climate/demo.js [hunk 2]
```diff
@@ -537,7 +547,59 @@ async function setupDashboard() {
...
+                        type: 'toggle',
+                        text: 'Fahrenheit',
+                        events: {
+                            click: function () {
+                                // Change temperature scale.
+                                if (temperatureScale === 'C') {
+                                    dataScope = 'TXF';
+                                    temperatureScale = 'F';
+                                } else {
+                                    dataScope = 'TXC';
```
Comment (sebastianbochan) [inline samples/dashboard/demos/climate/demo.js:585]: You can remove the `dataScope` from If/ELSE and place it after. 

```
let temperatureScale = temperatureScale === 'C' ? 'F' : 'C';

dataScope = 'TX' + temperatureScale;
```

File: samples/dashboard/demos/climate/demo.js [hunk 3]
```diff
@@ -623,8 +683,32 @@ async function setupDataPool() {
...
+    if (!cityDataTable.columns.TNC) {
+        cityDataTable.columns.TNC = cityDataTable.columns.TN.map(el =>
+            Highcharts.correctFloat((el - 273.15), 3)
+        );
+    }
+    if (!cityDataTable.columns.TNF) {
+        cityDataTable.columns.TNF = cityDataTable.columns.TN.map(el =>
+            Highcharts.correctFloat((el * (9 / 5) - 459.67), 3)
+        );
+    }
```
Comment (sebastianbochan) [inline samples/dashboard/demos/climate/demo.js:711]: Just an idea to replace a lot of IFs with a more generic code

```
const columns = ['TN', 'TX'];
const metric = ['C', 'F'];

columns.forEach(column => {
   metric.forEach(metric => {
        if (!cityDataTable.columns[column][metric]) {
             cityDataTable.columns[column][metric] = cityDataTable.columns[column].map(el =>
            Highcharts.correctFloat(
                 metric === 'C' ? (el - 273.15) : (el * (9 / 5) - 459.67), 
            3)
        }
   });
});


```

**Review by sebastianbochan — CHANGES_REQUESTED** [review]

File: samples/dashboard/demos/climate/demo.js [hunk 4]
```diff
@@ -623,8 +683,32 @@ async function setupDataPool() {
...
     }
+}
+
+// Calculate the average and max temperature in C and F from K.
+async function convertTemperature(city) {
+    const cityDataTable = (await dataPool.getStoreTable(city)).modified;
 
-    console.log(dataPool);
+    if (!cityDataTable.columns.TXC) {
+        cityDataTable.columns.TXC = cityDataTable.columns.TX.map(el =>
```
Comment (bre1470) [inline samples/dashboard/demos/climate/demo.js:693]: Use `DataTable.setColumns` and `const = DataTable.getColumn`.

**Review by bre1470 — CHANGES_REQUESTED** [review]

**Review by bre1470 — APPROVED** [review]
Thank you! Looks good. 👍

**Review by pawelfus — APPROVED** [review]
Thanks!

**Review by sebastianbochan — APPROVED** [review]

**Pull request merged**